### PR TITLE
[main][feat] release delta neutral

### PR DESCRIPTION
**Pull request by jr-alpaca**
## Description

- release Delta Neutral Vault
## ChangeLogs:

- DeltaNeutralOracle.sol
- DeltaNeutralVault.sol
- DeltaNeutralVaultConfig.sol
- DeltaNeutralVaultGateway.sol
- DeltaNeutralMdexWorker02.sol
- DeltaNeutralPancakeWorker02.sol

### CODE REVIEW

File: contracts/8.10/protocol/DeltaNeutralOracle.sol [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    // fair token0 amt: _sqrtK * sqrt(_px1/_px0)
+    // fair token1 amt: _sqrtK * sqrt(_px0/_px1)
+    // fair lp price = 2 * sqrt(_px0 * _px1)
+    // split into 2 sqrts multiplication to prevent uint overflow (note the 2**112)
+
+    uint256 _totalValueIn18;
+    {
+      uint8 padDecimals = 36 - (_d0 + _d1);
+      uint256 _totalValue = (((_sqrtK * 2 * (AlpacaMath.sqrt(_px0))) / (2**56)) * (AlpacaMath.sqrt(_px1))) / (2**56);
+      _totalValueIn18 = (_totalValue / (2**112)) * 10**(padDecimals); // change from 2**112 and 2**? to 2**18
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralOracle.sol:103]:
```suggestion
      _totalValueIn18 = (_totalValue / (2**112)) * 10**(padDecimals); // revert bumped up 2*112 from fdiv() and convert price to 1e18 unit
```

File: contracts/8.10/protocol/DeltaNeutralVault.sol [hunk 2]
```diff
@@ -0,0 +1,806 @@
...
+      uint256 _posId,
+      address _worker,
+      uint256 _principalAmount,
+      uint256 _borrowAmount,
+      uint256 _maxReturn,
+      bytes memory _workData
+    ) = abi.decode(_data, (address, uint256, address, uint256, uint256, uint256, bytes));
+
+    if (
+      OPENING != 1 &&
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVault.sol:768]: Please put comment here on why there should be this state.

File: contracts/8.10/protocol/DeltaNeutralVaultGateway.sol [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  error DeltaNeutralVaultGateway_UnTrustedPrice();
+  error DeltaNeutralVaultGateway_InsufficientReceive(
+    uint256 _stableAmount,
+    uint256 _assetAmount,
+    uint256 _minStableAmount,
+    uint256 _minAssetAmount
+  );
+
+  /// @dev constants
+  uint64 private constant BASIS_POINT = 10000;
```
Comment (alpanaca) [inline contracts/8.10/protocol/DeltaNeutralVaultGateway.sol:52]: Inconsistent naming

Comment (alpanaca) [reply contracts/8.10/protocol/DeltaNeutralVaultGateway.sol:52]:
```suggestion
  uint64 private constant MAX_BPS = 10000;
```

**Review by alpanaca — DISMISSED** [review]

**Review by alpanaca — APPROVED** [review]

**Review by alpanaca — APPROVED** [review]

**Review by spicysquid168 — APPROVED** [review]

**Pull request merged**